diff --git a/mk/kern.pre/mk b/mk/kern.pre.mk
index cda9efe77..0a76d51a 100644
--- a/mk/kern.pre.mk
+++ b/mk/kern.pre.mk
@@ -24,7 +24,7 @@ _MINUS_O= -O2
 endif
 ifeq (${MACHINE_CPUARCH},amd64)
 ifneq (${COMPILER_TYPE},clang)
-COPTFLAGS?=-O2 -fno-strict-aliasing -frename-registers -pipe -Wno-maybe-uninitialized #-finline-functions
+COPTFLAGS?=-O2 -fno-strict-aliasing -frename-registers -pipe -Wno-dangling-pointer -Wno-maybe-uninitialized #-finline-functions
 else
 COPTFLAGS?=-O2 -pipe
 endif